src/openeo_test_suite/lib/process_registry.py

```diff
@@ -63,15 +65,24 @@ def _load(self) -> Iterator[ProcessData]:
             raise ValueError(f"Invalid process test root directory: {self._root}")
         _log.info(f"Loading process definitions from {self._root}")
         for path in self._root.glob("*.json5"):
+            metadata_path = path.parent.parent / f"{path.stem}.json"
             try:
                 with path.open() as f:
                     data = json5.load(f)
                 if data["id"] != path.stem:
                     raise ValueError(
                         f"Process id mismatch between id {data['id']!r} and filename {path.name!r}"
                     )
+                # Metadata is stored in sibling json file
```

Comment:
There is not a json5 file for every json file yet, so you don't load all process metadata here. I guess this needs to be reversed to have the json files (from both root and proposals) as iterator and then load json5 if available.

Reply:
In the compariosn, we may also want to consider that some processes have been renamed and throw an error if they are still provided with the old name:

create_raster_cube -> create_cube
text_merge -> text_concat
load_result -> load_stac
debug -> inspect

Reply:
I did not write the process_selection code I just minimally adjusted it to include metadata, but assuming we adjust it to load .json first and supplement with json5 how would we handle process filtering with process_levels? Just exclude anything that has no level?

Reply:
Yeah, I wrote it. But for my use case. For your usecase we need to adapt it. Everything that has no level can be considered L4.

Reply:
This is done.